src/track/trackmetadatataglib.cpp

```diff
@@ -785,6 +846,46 @@ void readTrackMetadataFromXiphComment(TrackMetadata* pTrackMetadata,
                 toQStringFirstNotEmpty(tag.fieldListMap()["GROUPING"]));
     }
 
+    if (tag.fieldListMap().contains("TRACKNUMBER")) {
+        const QString trackNumberText(
+                toQStringFirstNotEmpty(tag.fieldListMap()["TRACKNUMBER"]));
+        // Try to parse both the track number and the optional total tracks
+        // from this field
+        const std::pair<TrackNumbers, TrackNumbers::ParseResult> trackNumbersFromString(
+                TrackNumbers::fromString(trackNumberText));
+        QString trackTotalText;
+        if ((TrackNumbers::ParseResult::VALID == trackNumbersFromString.second) &&
+                !trackNumbersFromString.first.hasTotal()) {
```

Comment:
I am not sure, but It looks like If a metadata contains "TOTALTRACKS" it should be the first choice. 
Probably related: http://forums.mp3tag.de/index.php?showtopic=16223